guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormViewImpl.java

```diff
@@ -93,87 +95,36 @@ public WellForm doUploadForm() {
         form.setEncoding( FormPanel.ENCODING_MULTIPART );
         form.setMethod( FormPanel.METHOD_POST );
 
-        final FileUpload up = new FileUpload( new Command() {
+        form.addSubmitHandler(new Form.SubmitHandler() {
             @Override
-            public void execute() {
-                showUploadingBusy();
-                form.submit();
+            public void onSubmit(final Form.SubmitEvent event) {
+                presenter.handleSubmit(event);
             }
-        } );
-        up.setName( HTMLFileManagerFields.UPLOAD_FIELD_NAME_ATTACH );
+        });
 
-        form.addSubmitHandler( new Form.SubmitHandler() {
+        uploader = new FileUpload(new Command() {
             @Override
-            public void onSubmit( final Form.SubmitEvent event ) {
-                String fileName = up.getFilename();
-                if ( fileName == null || "".equals( fileName ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.SelectFileUpload() );
-                    event.cancel();
-                } else if ( !( isValid( fileName ) ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.UnsupportedFileType() );
-                    event.cancel();
-                }
+            public void execute() {
+                form.submit();
```

Comment:
The only one EventHandler which wasn't moved to Presenter because the only action of this EventHandler is resend event to parent form.